app/lib/schemas/classicCVTemplateSchema.ts

```diff
@@ -1,50 +1,104 @@
-// /app/lib/schemas/classicCVTemplateSchema.ts
 import { z } from 'zod'
 
-// ContactType schema
-export const contactSchema = z.object({
-  phone: z.string(),
-  email: z.string().email(),
-  linkedin: z.string().optional(),
+// Schema for ExperienceType
+const ExperienceSchema = z.object({
+  companyName: z.string().describe('Company name'),
+  industry: z.string().describe('Industry of the company'),
+  speciality: z.string().describe('Speciality of the company'),
+  positionName: z.string().min(1).describe('Position name is required'),
+  description: z.string().describe('Description is required'),
+  date: z.string().describe('Date is required'),
 })
 
-// ExperienceType schema
-export const experienceSchema = z.object({
-  company: z.string(),
-  sector: z.string(),
-  companyType: z.string(),
-  position: z.string(),
-  duration: z.string(),
-  description: z.string(),
-})
-
-// EducationType schema
-export const educationSchema = z.object({
-  institution: z.string(),
-  degree: z.string(),
-  fieldOfStudy: z.string(),
-  duration: z.string(),
-})
-
-// CertificateType schema
-export const certificateSchema = z.object({
-  certTitle: z.string(),
-  certDate: z.string(),
-})
+// Schema for Education
+const EducationSchema = z
+  .object({
+    schoolName: z
+      .string()
+      .optional()
+      .describe('do not fill if not provided in original cv in prompt.'),
+    studyName: z
+      .string()
+      .optional()
+      .describe('do not fill if not provided in original cv in prompt.'),
+    year: z
+      .string()
+      .optional()
+      .describe('do not fill if not provided in original cv in prompt.'),
+  })
+  .optional()
+  .describe('do not fill if not provided in original cv in prompt.')
 
-// ClassicCVTemplate schema
+// Main schema for CvDataType
 export const classicCVTemplateSchema = z.object({
-  name: z.string(),
-  positionTitle: z.string(),
-  portfolio: z.string().optional(),
-  contact: contactSchema,
-  technologies: z.array(z.string()),
-  summary: z.string(),
-  experience: z.array(experienceSchema),
-  education: z.array(educationSchema),
-  skills: z.array(z.string()),
-  certificates: z.array(certificateSchema),
-  company: z.string(),
+  name: z.string().min(1).describe('Name item cannot be empty. your name'),
+  positionTitle: z
+    .string()
+    .min(1)
+    .describe(
+      "Position item cannot be empty. Name of the position yo apply for. generic. do not fill the same if it is specified too much. for example if 'Software Engineer (Node.js) - AI Solutions' return 'Software Engineer'",
+    ),
```

Comment:
**praise**: Excellent use of Zod for schema validation, ensuring data integrity and reducing potential errors.